### PR TITLE
expose emulator debug log

### PR DESCRIPTION
**Pull request by bjartek**
Expose https://github.com/onflow/flow-emulator/pull/166 in CLI

### CODE REVIEW

**Comment by bjartek** [comment]
The edit replace line in go.mod will be removed once emulator pr is merged

**Comment by bjartek** [comment]
Not really sure on how to test this TBH, i cannot find existing tests that call the method we added a new option to here.

File: pkg/flowkit/gateway/emulator.go [hunk 1]
```diff
@@ -38,10 +39,11 @@ import (
...
 type EmulatorGateway struct {
-	emulator *emulator.Blockchain
-	backend  *backend.Backend
-	ctx      context.Context
-	logger   *logrus.Logger
+	emulator      *emulator.Blockchain
+	backend       *backend.Backend
+	ctx           context.Context
+	logger        *logrus.Logger
+	emultorLogger *zerolog.Logger
```
Comment (sideninja) [inline pkg/flowkit/gateway/emulator.go:46]:
```suggestion
	emulatorLogger *zerolog.Logger
```

Comment (sideninja) [inline pkg/flowkit/gateway/emulator.go:46]: These loggers should be combined, but this is probably work for another PR.

File: go.mod [hunk 2]
```diff
@@ -25,3 +25,5 @@ require (
 	gonum.org/v1/gonum v0.11.0
 	google.golang.org/grpc v1.45.0
 )
+
+replace github.com/onflow/flow-emulator => github.com/findonflow/flow-emulator custom-zerolog
```
Comment (sideninja) [inline go.mod:29]: Let me know once emulator PR is in and this is removed so I approve the PR.

**Comment by sideninja** [comment]
LMK when emulator PR is in and make sure tests and lint passes

File: pkg/flowkit/gateway/emulator.go (hunk 1 above)
Comment (bjartek) [reply pkg/flowkit/gateway/emulator.go]: Agreed

Comment (psiemens) [reply pkg/flowkit/gateway/emulator.go]: Also agreed 😄

File: pkg/flowkit/gateway/emulator.go [hunk 3]
```diff
@@ -54,14 +56,16 @@ func NewEmulatorGateway(serviceAccount *flowkit.Account) *EmulatorGateway {
 
 func NewEmulatorGatewayWithOpts(serviceAccount *flowkit.Account, opts ...func(*EmulatorGateway)) *EmulatorGateway {
 
+	defaultEmulatorLogger := zerolog.Nop()
 	gateway := &EmulatorGateway{
-		ctx:    context.Background(),
-		logger: logrus.New(),
+		ctx:           context.Background(),
+		logger:        logrus.New(),
+		emultorLogger: &defaultEmulatorLogger,
```
Comment (psiemens) [inline pkg/flowkit/gateway/emulator.go:63]:
```suggestion
		emulatorLogger: &defaultEmulatorLogger,
```

File: pkg/flowkit/gateway/emulator.go [hunk 4]
```diff
@@ -54,14 +56,16 @@ func NewEmulatorGateway(serviceAccount *flowkit.Account) *EmulatorGateway {
...
 	}
 	for _, opt := range opts {
 		opt(gateway)
 	}
-	gateway.emulator = newEmulator(serviceAccount)
+	gateway.emulator = newEmulator(serviceAccount, gateway.emultorLogger)
```
Comment (psiemens) [inline pkg/flowkit/gateway/emulator.go:68]:
```suggestion
	gateway.emulator = newEmulator(serviceAccount, gateway.emulatorLogger)
```

File: pkg/flowkit/gateway/emulator.go [hunk 5]
```diff
@@ -74,6 +78,12 @@ func WithLogger(logger *logrus.Logger) func(g *EmulatorGateway) {
 	}
 }
 
+func WithEmulatorLogger(logger *zerolog.Logger) func(g *EmulatorGateway) {
+	return func(g *EmulatorGateway) {
+		g.emultorLogger = logger
```
Comment (psiemens) [inline pkg/flowkit/gateway/emulator.go:83]:
```suggestion
		g.emulatorLogger = logger
```

**Review by psiemens — COMMENTED** [review]
Great idea to add this. 👍

File: go.mod (hunk 2 above)
Comment (bjartek) [reply go.mod:29]: emulator PR is now in but emulator has not been release yet AFAIK @sideninja

Comment (srinjoyc) [reply go.mod:29]: Hey, @bjartek - @sideninja is away for a few days but coming back on the 14th, so he should be able to prioritize this once he's back.

**Comment by bjartek** [comment]
I opend up a new PR.

**Pull request closed without merging**